core-bundle/contao/forms/Form.php

```diff
@@ -560,10 +587,41 @@ protected function processFormData($arrSubmitted, $arrLabels, $arrFields, $arrFi
 			System::getContainer()->get('monolog.logger.contao.forms')->info('Form "' . $this->title . '" has been submitted by a guest.');
 		}
 
+		$targetPageData = null;
+
 		// Check whether there is a jumpTo page
 		if (($objJumpTo = $this->objModel->getRelated('jumpTo')) instanceof PageModel)
 		{
-			$this->jumpToOrReload($objJumpTo->row());
+			$targetPageData = $objJumpTo->row();
+		}
+
+		// Set the confirmation message, if any
+		if ($this->objModel->confirmationMessage)
+		{
+			$confirmationMessage = $this->objModel->confirmationMessage;
+			$confirmationMessage = System::getContainer()->get('contao.string.simple_token_parser')->parse($confirmationMessage, $arrSubmitted);
+			$confirmationMessage = System::getContainer()->get('contao.insert_tag.parser')->replaceInline($confirmationMessage);
```

Comment:
Should we replace this in reverse order to ensure that no insert-tag gets hacked into the message with the simple tokens?

Reply:
Not sure. On the other hand, the form data may be used in the insert tags like `{{email::##form_email##}}`.

Reply:
`$arrSubmitted` seems to be input-encoded so it should probably be fine. But for this uses case I think it would make sense to do something like `array_map(StringUtil::specialchars(...), $arrSubmitted)` to make this safe even if a form widget is configured incorrectly.

Reply:
Implemented in 185f2893.